kernel/src/process.rs

```diff
@@ -178,44 +183,51 @@ pub fn load_processes<C: Chip>(
                 }
             };
 
-            // Now we can get a slice which only encompasses the app. At this
-            // point, since the version number in the beginning of the header is
-            // valid, we consider further parsing errors to be actual errors and
-            // report them to the caller.
+            // Now we can get a slice which only encompasses the app. We will
+            // either parse this as an actual app, or skip over this region.
             let app_flash = remaining_flash
                 .get(0..app_length as usize)
                 .ok_or(ProcessLoadError::NotEnoughFlash)?;
 
-            // Try to create a process object from that app slice.
-            let (process, memory_offset) = Process::create(
-                kernel,
-                chip,
-                app_flash,
-                header_length as usize,
-                version,
-                app_memory_ptr,
-                app_memory_size,
-                fault_response,
-                i,
-            )?;
-
-            // Check to see if actually got a valid process to execute. If we
-            // didn't and we didn't get a loading error (aka we got to this
-            // point), then the app is a disabled process or just padding.
-            if process.is_some() {
-                if config::CONFIG.debug_load_processes {
-                    debug!(
-                        "Loaded process[{}] from flash=[{:#010X}:{:#010X}] into sram=[{:#010X}:{:#010X}] = {:?}",
-                        i,
-                        app_flash.as_ptr() as usize,
-                        app_flash.as_ptr() as usize + app_flash.len(),
-                        app_memory_ptr as usize,
-                        app_memory_ptr as usize + memory_offset,
-                        process.map(|p| p.get_process_name())
-                    );
+            // Check if we should continue parsing this app. If the initial
+            // header parsing failed (i.e. the header_length is 0), we don't
+            // bother.
```

Comment:
Something about either the logic or the phrasing here is confusing.

If "initial header parsing failed", shouldn't `parse_tbf_header_lengths` have returned an error and we would never get here?

Reply:
`parse_tbf_header_lengths` can fail for basically two reasons: either we are at the end of apps or there is a corrupt app header. If we are at the end of apps we have returned already above. This check differentiates a successful initial header parse from a corrupt header.

Reply:
I just pushed some comment and name updates that helps clarify what was confusing me. The key for me was making explicit a tbf entry, that is the region of flash described by a tbf header, which may or may not actually be an app (I was really hung up on why there was any meaningful "app_flash" in the case where it's not a valid app).